### PR TITLE
Fasten dump, fix yaxis limits/inversion in wiggle plot

### PR DESCRIPTION
**Pull request by nikita-klsh**

### CODE REVIEW

File: seismicpro/src/gather/gather.py [hunk 1]
```diff
@@ -1435,7 +1434,9 @@ def _plot_wiggle(self, ax, title, x_ticker, y_ticker, x_tick_src=None, y_tick_sr
 
         patch = PathPatch(Path(verts, codes), color=color, alpha=alpha)
         ax.add_patch(patch)
-        ax.invert_yaxis()
+        ax.update_datalim([(0, 0), (0, traces.shape[1])])
```
Comment (alexanderkuvaev) [inline seismicpro/src/gather/gather.py:1437]: Shouldn't it be
```suggestion
        ax.update_datalim([(0, 0), traces.shape)])
```

File: seismicpro/src/gather/gather.py [hunk 2]
```diff
@@ -366,10 +363,12 @@ def dump(self, path, name=None, retain_parent_segy_headers=True):
...
             dump_handler.trace = self.data
-            for i, dump_h in trace_headers_dict.items():
+            for i, trace_headers in enumerate(self[used_header_names]):
                 if retain_parent_segy_headers:
-                    dump_handler.header[i].update(parent_handler.header[trace_ids[i]])
-                dump_handler.header[i].update({**dump_h, segyio.TraceField.TRACE_SAMPLE_INTERVAL: sample_rate})
+                    dump_handler.header[i] = parent_handler.header[trace_ids[i]]
+                dump_handler.header[i].update({**dict(zip(used_header_bytes, trace_headers)),
+                                               segyio.TraceField.TRACE_SAMPLE_INTERVAL: sample_rate,
+                                               segyio.TraceField.TRACE_SEQUENCE_FILE: i})
```
Comment (alexanderkuvaev) [inline seismicpro/src/gather/gather.py:371]:
```suggestion
                                               segyio.TraceField.TRACE_SEQUENCE_FILE: i+1})
```
TSF starts with 1, strange that it worked

File: seismicpro/src/gather/gather.py [hunk 3]
```diff
@@ -366,10 +363,12 @@ def dump(self, path, name=None, retain_parent_segy_headers=True):
 
             # Dump traces and their headers. Optionally copy headers from the parent SEG-Y file.
             dump_handler.trace = self.data
-            for i, dump_h in trace_headers_dict.items():
+            for i, trace_headers in enumerate(self[used_header_names]):
                 if retain_parent_segy_headers:
-                    dump_handler.header[i].update(parent_handler.header[trace_ids[i]])
-                dump_handler.header[i].update({**dump_h, segyio.TraceField.TRACE_SAMPLE_INTERVAL: sample_rate})
+                    dump_handler.header[i] = parent_handler.header[trace_ids[i]]
```
Comment (alexanderkuvaev) [inline seismicpro/src/gather/gather.py:368]: Please check that parent segy data is not affected by the following update

File: seismicpro/src/gather/gather.py [hunk 4]
```diff
@@ -337,20 +337,17 @@ def dump(self, path, name=None, retain_parent_segy_headers=True):
...
 
-        trace_headers = self.headers.reset_index()
         sample_rate = np.int32(self.sample_rate * 1000) # Convert to microseconds
-        # Remember ordinal numbers of traces in the parent SEG-Y file to further copy their headers
-        # and reset them to start from 1 in the resulting file to match SEG-Y standard.
-        trace_ids = trace_headers["TRACE_SEQUENCE_FILE"].values - 1
-        trace_headers["TRACE_SEQUENCE_FILE"] = np.arange(len(trace_headers)) + 1
+        if retain_parent_segy_headers:
+            # Remember ordinal numbers of traces in the parent SEG-Y file to further copy their headers
+            trace_ids = self["TRACE_SEQUENCE_FILE"].ravel() - 1
```
Comment (alexanderkuvaev) [inline seismicpro/src/gather/gather.py:343]: Let's remove `if` and calculate `trace_ids` unconditionally

**Review by alexanderkuvaev — CHANGES_REQUESTED** [review]

File: seismicpro/src/gather/gather.py (hunk 3 above)
Comment (nikita-klsh) [reply seismicpro/src/gather/gather.py:368]: It's not

**Review by alexanderkuvaev — APPROVED** [review]

**Pull request merged**